Mathlib/AlgebraicTopology/SimplicialSet/StrictSegal.lean

```diff
@@ -86,6 +87,151 @@ theorem spineToSimplex_edge (f : Path X n) (j l : ℕ) (hjl : j + l ≤ n) :
+instance : Quasicategory X := by
+  apply quasicategory_of_filler X
+  intro n i σ₀ h₀ hₙ
+  exists spineToSimplex <| Path.map (horn.spineId i h₀ hₙ) σ₀
+  intro j hj
+  apply spineInjective
+  ext k
+  · simp only [spineEquiv, spine_arrow, Function.comp_apply, Equiv.coe_fn_mk]
+    rw [← types_comp_apply (σ₀.app _) (X.map _), ← σ₀.naturality]
+    let ksucc := k.succ.castSucc
+    have hkj : ksucc < j ∨ j < ksucc ∨ j = ksucc := by omega
+    rcases hkj with hlt | hgt | heq
+    · rw [← spine_arrow, spine_δ_arrow_lt _ hlt]
+      simp only [Path.map, spine_arrow, Fin.coe_eq_castSucc]
+      apply congr_arg
+      simp only [horn, horn.spineId, standardSimplex, uliftFunctor, Functor.comp_obj,
+        yoneda_obj_obj, whiskering_obj_obj_map, uliftFunctor_map, yoneda_obj_map,
+        standardSimplex.objEquiv, Equiv.ulift, Equiv.coe_fn_symm_mk,
+        Quiver.Hom.unop_op, horn.face_coe, Subtype.mk.injEq]
+      rw [mkOfSucc_δ_lt hlt]
+      rfl
+    · rw [← spine_arrow, spine_δ_arrow_gt _ hgt]
+      simp only [Path.map, spine_arrow, Fin.coe_eq_castSucc]
+      apply congr_arg
+      simp only [horn, horn.spineId, standardSimplex, uliftFunctor, Functor.comp_obj,
+        yoneda_obj_obj, whiskering_obj_obj_map, uliftFunctor_map, yoneda_obj_map,
+        standardSimplex.objEquiv, Equiv.ulift, Equiv.coe_fn_symm_mk,
+        Quiver.Hom.unop_op, horn.face_coe, Subtype.mk.injEq]
+      rw [mkOfSucc_δ_gt hgt]
+      rfl
+    · /- The only inner horn of `Δ[2]` does not contain the diagonal edge. -/
+      have hn0 : 0 < n := by
+        apply Nat.pos_of_ne_zero
+        intro h1; subst h1
+        have hk : k = 0 := by omega
+        have hi : i = 1 := by fin_cases i <;> (try contradiction); rfl
+        subst hi hk
+        exact hj heq
+      /- We construct the triangle in the standard simplex as a 2-simplex in
+      the horn. While the triangle is not contained in the inner horn `Λ[2, 1]`,
+      we can inhabit `Λ[n + 2, i] _[2]` by induction on `n`. -/
+      let triangle : Λ[n + 2, i] _[2] := by
+        cases n with
+        | zero => contradiction
+        | succ _ => exact horn.primitiveTriangle i h₀ hₙ k (by omega)
+      /- The interval spanning from `k` to `k + 2` is equivalently the spine
+      of the triangle with vertices `k`, `k + 1`, and `k + 2`. -/
+      have hi : ((horn.spineId i h₀ hₙ).map σ₀).interval k 2 (by omega) =
+          X.spine 2 (σ₀.app _ triangle) := by
+        ext m
+        simp [spine_arrow, Path.interval, Path.map]
+        rw [← types_comp_apply (σ₀.app _) (X.map _), ← σ₀.naturality]
+        apply congr_arg
+        simp only [horn, standardSimplex, uliftFunctor, Functor.comp_obj,
+          whiskering_obj_obj_obj, yoneda_obj_obj, uliftFunctor_obj, ne_eq,
+          whiskering_obj_obj_map, uliftFunctor_map, yoneda_obj_map, len_mk,
+          Nat.reduceAdd, Quiver.Hom.unop_op]
+        cases n with
+        | zero => contradiction
+        | succ _ => fin_cases m <;> (ext x; fin_cases x <;> rfl)
```

Comment:
If you want, you can reorder the case split on `x` and `m`, by starting with `ext x`. In that case you only make one call to `ext`. It also allows you to drop the `()`.
```suggestion
        | succ _ => ext x; fin_cases x <;> fin_cases m <;> rfl
```
But if you prefer the other order, that's fine with me.